stock-tracker/src/features/search/redux/searchMiddlware.test.ts

```diff
@@ -0,0 +1,91 @@
+import {
+  SearchSocketServiceDependency,
+  searchMiddleware
+} from "./searchMiddleware";
+import { Store, Dispatch } from "redux";
+import { AppState } from "store/rootReducer";
+import { ADD_SYMBOL, ADD_SEARCH_INPUT } from "./actionTypes";
+import { setChartLoadingAction } from "features/chart";
+import { setLoadingNewsAction } from "features/latestNews";
+import { setLoadingKeyStatsAction } from "features/keyStats";
+import { setLoadingOverviewAction } from "features/overview";
+import { setLoadingPeersAction } from "features/topPeers";
+
+describe("Testing Search Middleware", () => {
+  let mockSocket: SearchSocketServiceDependency;
+  let emit: jest.Mock;
+  let store: Pick<Store<Pick<AppState, "chart">>, "getState" | "dispatch">;
+  let next: Dispatch;
+  let action: { type: string; payload: string };
+  let dispatch: jest.Mock;
+
+  beforeEach(() => {
+    emit = jest.fn();
+    dispatch = jest.fn();
+
+    mockSocket = {
+      socketService: {
+        create: () => ({
+          emit
+        })
+      }
+    };
+
+    store = {
+      getState: () => ({
+        chart: { chartTime: "1Y", chartData: [], loading: false, error: false }
+      }),
+      dispatch
+    };
+    next = jest.fn();
+  });
+
+  describe("when called with a ADD_SYMBOL action", () => {
+    it("should call the emit, next and dispatch functions", () => {
+      action = {
+        type: ADD_SYMBOL,
+        payload: "AAPL"
+      };
+      searchMiddleware(mockSocket)(store)(next)(action);
+      expect(emit).toHaveBeenCalledTimes(1);
+      expect(next).toHaveBeenCalledTimes(1);
+      expect(next).toHaveBeenCalledWith(action);
+      expect(dispatch).toHaveBeenCalledTimes(5);
+    });
+    it("should call emit with the correct data", () => {
+      const SYMBOL_INPUT = "symbol";
+      searchMiddleware(mockSocket)(store)(next)(action);
+      expect(emit).toHaveBeenCalledWith(
+        SYMBOL_INPUT,
+        action.payload,
+        store.getState().chart.chartTime
+      );
+    });
+    it("should call dispatch with the correct actions", () => {
```

Comment:
Q: would you not want to split these into their own tests in case you want to test whether they are dispatched with the correct payloads in the future?

Reply:
I didn't think it was necessary as these loading actions do not contain a payload, only a type

Reply:
I think u should have a separate PR for searchMiddleware